### ISSUE TITLE
File Share: cannot disable compression 

### ISSUE THREAD

**Issue by woeisme** [micahflee/onionshare#1328]
[according to documentation ](https://docs.onionshare.org/2.3.1/en/features.html?highlight=compression)

>Also, if you uncheck this box, people will be able to download the individual files you share rather than a single compressed version of all the files.

but unchecking this box does not disable compression, for large files this eats up all space in /tmp (as documented [here](https://github.com/micahflee/onionshare/issues/1114) before crashing. 

OnionShare version 2.3.1

**Comment by mig5** [comment]
'rather than a single compressed version of all the files' - means 'a single zip file'.

It's true that in website mode, a *gzipped* version of each file is created in `/tmp/` - which is your issue, but isn't the same as what the documentation is talking about.

I thought we had an issue open about it already but can't find it - so I'll rename this 'Allow ability to serve website files directly rather than gzipped copies'

**Comment by mig5** [comment]
Relates to https://github.com/micahflee/onionshare/pull/1256 - maybe it would be simpler to not 're-read' (and re-compress gzip) files but serve them directly from the original source.

Of course, people will then complain that it's slower to load files (because of the lack of gzip compression).. can't win.

**Comment by whew** [comment]
I think it's a good idea to serve user files uncompressed. It's a lot simpler and would actually solve a lot of issues.

Pros:
* solve the large file problem (#1114)
* pause/resume downloads in Tor Browser (#1333)
* reflect changes to files immediately (#1059)
* solve the bug I described [here](https://github.com/micahflee/onionshare/pull/1256#issuecomment-749350720) (tl;dr: if a file is modified after it's been compressed by onionshare, then `curl --compressed` gets the gzipped [old] version and `curl` gets the modified version).
* no extra copies of files on disk: when we delete temporary gzip files, we use python's `os.remove` which means they can be recovered using data recovery tools e.g. [photorec](https://en.wikipedia.org/wiki/PhotoRec) (related to #1205)
* no time wasted compressing already compressed files, e.g. `jpg`, `pdf`, `mp4`, `doc`, etc.

Cons:
* compressible files will take longer to transfer e.g. `txt`, `iso`

**Comment by mig5** [comment]
Yes I came to the same conclusion seeing your other update. I think I agree! 

There are other things we are looking at to help speed up downloads which are more to do with how Tor cells work https://github.com/micahflee/onionshare/issues/1295 - so with my most optimistic hat on, the 'con' might not end up being that big a deal in the end after all, if we achieve that.

**Comment by whew** [comment]
> reflect changes to files immediately (#1059)

This would solve #1059 in every case except share mode with at least 2 files. In that case, we zip all the files before we start the server, so any subsequent changes to the files on disk won't be reflected. My pull request #1256 was basically done except for this part, I'm not sure what should be done about it.
